src/user/mod.rs

```diff
@@ -121,26 +147,32 @@ pub enum UserAction {
     AddInput,
     #[display("Please select an input to remove.")]
     RemoveInput,
-    GenLib,
     #[display("Is the input a flake?")]
     IsInputFlake,
-    #[display("placeholder")] // TODO write display for this
-    GenBin(Lang),
     #[display("Encountered an error: {0}")]
     Error(String),
+    #[display("{0}")]
+    Rust(rust::Action),
 }
 
-#[derive(Eq, PartialEq, Debug, Copy, Clone, Display)]
-pub enum Lang {
+#[derive(Eq, PartialEq, Debug, Copy, Clone, Display, FromStr)]
+pub(crate) enum Lang {
+    #[display("rust")]
     Rust,
+    #[display("haskell")]
     Haskell,
+    #[display("python")]
     Python,
+    #[display("javascript")]
     JavaScript,
 }
 
-pub fn query_user_input(prompt: Vec<String>, items: Vec<String>, files: bool) -> String {
+pub(crate) fn query_user_input(
+    prompt: Vec<String>,
+    items: Vec<String>,
```

Comment:
I would suggest, instead of giving this function `Vec`s as input, use Iterators, avoiding unnecessary allocations.
```suggestion
    prompt: impl Iterator<Item = &str>,
    items: impl Iterator<Item = &str>,
```